src/WebJobs.Script/Description/CSharp/FunctionAssemblyLoader.cs

```diff
@@ -45,21 +46,28 @@ private void Dispose(bool disposing)
             }
         }
 
-        private Assembly ResolveAssembly(object sender, ResolveEventArgs args)
+        internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
         {
             FunctionAssemblyLoadContext context = GetFunctionContext(args.RequestingAssembly);
             Assembly result = null;
 
             if (context != null)
             {
                 result = context.ResolveAssembly(args.Name);
+
+                // Failed to resolve a function assembly dependency, log the failure as this
+                // is usually caused by missing private assemblies.
+                if (result == null)
+                {
+                    context.TraceWriter.Warning(string.Format(CultureInfo.InvariantCulture, "Unable to find assembly '{0}'. Are you missing a private assembly file?", args.Name));
```

Comment:
Some additional logging here to enhance the experience when users miss assemblies. This will give them details, at the function level, about the assemblies we failed to resolve.